models/migrations/migrations.go

```diff
@@ -737,3 +741,39 @@ func dropTableColumns(sess *xorm.Session, tableName string, columnNames ...strin
 
 	return nil
 }
+
+// modifyColumn will modify column's type or other propertity. SQLITE is not supported
+func modifyColumn(x *xorm.Engine, tableName string, col *schemas.Column) error {
+	var indexes map[string]*schemas.Index
+	var err error
+	// MSSQL have to remove index at first, otherwise alter column will fail
+	// ref. https://sqlzealots.com/2018/05/09/error-message-the-index-is-dependent-on-column-alter-table-alter-column-failed-because-one-or-more-objects-access-this-column/
+	if x.Dialect().URI().DBType == schemas.MSSQL {
+		indexes, err = x.Dialect().GetIndexes(x.DB(), context.Background(), tableName)
+		if err != nil {
+			return err
+		}
+
+		for _, index := range indexes {
+			_, err = x.Exec(x.Dialect().DropIndexSQL(tableName, index))
```

Comment:
If we're dropping index and then re-creating it then we might as well just do recreatetable, there will be absolutely zero difference in performance between two.

Reply:
That won't be same and also this is only for MSSQL that is actually not so commonly used imho

Reply:
@CirnoT I think recreate indexes should have better performance than recreate the table. Recreate table contains insert data and generate index, recreate indexes will contain drop index and generate index. So we could compare drop index time and insert data time.

@zeripath Just got an idea. And I think recreatetable could have a performance optimization, we should copy data from original table and then create indexes but not create indexes and then copy data from original.

Reply:
> @zeripath Just got an idea. And I think recreatetable could have a performance optimization, we should copy data from original table and then create indexes but not create indexes and then copy data from original.

I think that might not be allowed on every db - I'm sure I tried it that way round too.

Reply:
@zeripath It should be allowed. Sometimes, this is a tricky to insert plenty of data to database. A notice is we have to manually confirm no duplicated content on unique index columns when inserting data.

Reply:
@lunny @lafriks This is still pointless, the `ALTER TABLE` statement here will cause `copy to tmp` on InnoDB engine anyway, so you are in fact doing what recreate table does, just in more lines of code for absolutely zero benefit on performance.